### PR TITLE
Route53 routing policy

### LINKED ISSUE #207: add support for routing policy to route53 module
##### Issue Type:

Feature Idea
##### Ansible Version:

ansible 1.7.2
##### Environment:

Ubuntu 14.04
##### Summary:

route53 module does not let you set routing policy. http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-policy.html

(Filing on behalf of user).

### PR DESCRIPTION
**Pull request by zimbatm**
Adds support for routing policies by using the underlying boto facilities. Fixes #207:

Additional work:
- [x] Improve change-detection to support all the extra attributes (by comparing record.to_xml())
- [x] Improve the record updates to use UPSERT instead of DELETE+CREATE
- ~~[ ] Add a route53_healthcheck module to describe these resources.~~ (see #1276)

### CODE REVIEW

**Comment by bennojoy** [comment]
@zimbatm could you please resolve the conflicts and also squash the commits to a single one

**Comment by zimbatm** [comment]
@bennojoy why do you want me to squash the commits ? They are all different logical changes

File: cloud/amazon/route53.py [hunk 1]
```diff
@@ -93,6 +93,46 @@
     required: false
     default: false
     version_added: "1.9"
+  identifier:
```
Comment (bcoca) [inline cloud/amazon/route53.py:96]: missing version_added on all new options

File: cloud/amazon/route53.py [hunk 2]
```diff
@@ -93,6 +93,46 @@
...
+    description:
+      - Weighted and latency-based resource record sets only. An identifier
+        that differentiates among multiple resource record sets that have the
+        same combination of DNS name and type.
+    required: false
+    default: null
+    aliases: []
```
Comment (bcoca) [inline cloud/amazon/route53.py]: no need for empty aliases

**Comment by bennojoy** [comment]
@zimbatm  just one squashed commit because that keeps the history concise and makes it easy in  reviewing and merging code.. thanks..

File: cloud/amazon/route53.py (hunk 1 above)
Comment (zimbatm) [inline cloud/amazon/route53.py:96]: ~~which version should I been using ?~~ EDIT: using "2.0"

**Comment by zimbatm** [comment]
@bennojoy alright. squashed, rebased and comments addressed

**Comment by bennojoy** [comment]
@zimbatm  seems to have a conflict creedped , can you please resolve .. thanks

**Comment by zimbatm** [comment]
@bennojoy  fixed

**Comment by bennojoy** [comment]
@zimbatm  merged thanks

**Comment by zimbatm** [comment]
@bennojoy do you mind taking a look at the associated #1276 module as well ? Without being able to create new healthchecks this PR is not super useful yet.

**Pull request merged**